libazureinit/src/provision/ssh.rs

```diff
@@ -101,10 +143,38 @@ mod tests {
                 }
             }
         }
-
         Ok(None)
     }
 
+    fn mock_update_sshd_config(sshd_config_path: &PathBuf) -> io::Result<()> {
```

Comment:
With the above suggestion about providing the path to the function, you can drop this mock and write your tests against the real function

Reply:
Done